### PR TITLE
test/calico: initial commit

### PR DESCRIPTION
**Pull request by invidian**
This PR adds initial test for Calico, which checks, that each Node
object in the cluster has associated HostEndpoint Calico object, which
ensures, that GlobalNetworkPolicy objects take effect.

Refs #137.

Signed-off-by: Mateusz Gozdek <mateusz@kinvolk.io>

NOTE: This PR also includes commits from #209, which should be merged first.

### CODE REVIEW

File: test/components/cluster-autoscaler/cluster-autoscaler_test.go [hunk 1]
```diff
@@ -31,12 +31,7 @@ const (
 )
 
 func TestClusterAutoscalerDeployments(t *testing.T) {
-	client, err := testutil.CreateKubeClient(t)
-	if err != nil {
-		t.Errorf("could not create Kubernetes client: %v", err)
-	}
-
-	t.Log("got kubernetes client")
+	client := testutil.CreateKubeClient(t)
```
Comment (surajssd) [inline test/components/cluster-autoscaler/cluster-autoscaler_test.go]: This needs rebase?

Comment (invidian) [reply test/components/cluster-autoscaler/cluster-autoscaler_test.go]: Yes, thanks. I've done it already, but I didn't push the changes yet :smile:

**Comment by invidian** [comment]
I'll trigger the CI again to make sure the tests are not flaky.

**Comment by invidian** [comment]
@rata thanks for the review. I think at this point, testing YAML content would be more complex to implement, so I'd rather not call it "unit" test. In my opinion, the test I implemented provides good balance between the complexity of implementation and the coverage of the functionality. I agree that it doesn't test every aspect of the solution, but at least it should catch the regression which could be caused by the way we create those objects. If Calico or Packet changes the behavior, then we won't notice it, but this should probably be covered by blackbox tests we don't have right now.

**Comment by invidian** [comment]
>Do you think it makes sense to remove this test once we have the blackbox test? Maybe we can point it out in the blackbox issue. And, hopefully, have unit tests later for things that will be way faster?

I guess that depends on how do we implement blackbox tests.

**Comment by invidian** [comment]
Except regular CI flakiness, the test itself seems stable.

**Pull request merged**